domain/model/mesh_results.py

```diff
@@ -37,8 +41,14 @@ def get_alias(self, agent_id: AgentID) -> str:
             return f"[agent_id={agent_id} not found]"
         return agent.alias
 
-    def insert(self, agent: Agent) -> None:
-        self._agents[agent.id] = agent
+    def list_agent_ids(self) -> List[AgentID]:
+        return list(self._agents)
+
+    def filter(self, agent_ids: List[AgentID]):
+        result = Agents()
+        for agent_id in agent_ids:
+            result.insert(deepcopy(self.get_by_id(agent_id)))
```

Comment:
Why do we need the `deepcopy` here?

Reply:
I deepcopy to guard from the situation when other parts of code mutate the agent in filtered Agents and in consequence mutate it also in cached Agents. After second thought, it is very unlikely, so I'll remove that.

Reply:
I agree that list/filter functions should return a copy to prevent mutating data by mistake.

Reply:
The list of agents and their properties should be for all practical purpose read-only after it is constructed. Can we achieve that? If we change the code back to the Agents instance always holding only agents used in the mesh, we do not need the filter method. Anyway, this is non-critical aspect.